test/test_um2netcdf.py

```diff
@@ -70,46 +70,86 @@ def mule_vars(z_sea_rho_data, z_sea_theta_data):
     return um2nc.MuleVars(um2nc.GRID_NEW_DYNAMICS, d_lat, d_lon, z_sea_rho_data, z_sea_theta_data)
 
 
-def set_default_attrs(cube, item_code: int, var_name: str):
-    """Add subset of default attributes to flesh out cube like objects."""
-    cube.__dict__.update({"item_code": item_code,
-                          "var_name": var_name,
-                          "long_name": "",
-                          "coord": {"latitude": 0.0,  # TODO: real val = ?
-                                    "longitude": 0.0},  # TODO: real val
-                          "cell_methods": [],
-                          "data": None,
-                          })
+@dataclass(frozen=True)
+class DummyStash:
+    """
+    Partial Stash representation for testing.
+    """
+    section: int
+    item: int
 
-    section, item = um2nc.to_stash_code(item_code)
-    cube.attributes = {um2nc.STASH: DummyStash(section, item)}
 
+class DummyCube:
+    """
+    Imitation iris Cube for unit testing.
+    """
 
-@pytest.fixture
-def air_temp_cube():
-    # data copied from aiihca.paa1jan.subset file
-    name = "air_temperature"
-    m_air_temp = mock.NonCallableMagicMock(spec=iris.cube.Cube, name=name)
-    set_default_attrs(m_air_temp, 30204, name)
-    return m_air_temp
+    def __init__(self, item_code, var_name=None, attributes=None,
+                 units=None, coords=None):
+        self.item_code = item_code
+        self.var_name = var_name or "unknown_var"
+        self.attributes = attributes
```

Comment:
TODO: add the attribute fixes (e.g. `self.attrs = attrs or {}`

Reply:
I've raised this in #120 to defer `DummyCube` refactoring to a separate block of work (as this PR is deduplicating cube like test objs).